lib/k8s-e2e-tests/src/metrics.rs

```diff
@@ -91,6 +93,28 @@ pub async fn wait_for_vector_started(
     Ok(())
 }
 
+/// This helper function performs an HTTP request to the specified URL and
+/// validates the presence of the host metrics.
+pub async fn assert_host_metrics_present(url: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let metrics = load(url).await?;
+    let mut required_metrics: HashSet<_> = vec![
+        "host_load1",
+        "host_load5",
+        "host_cpu_seconds_total",
+        "host_filesystem_total_bytes",
```

Comment:
Did we ever determine whether or not this actually validates from the host?

Reply:
These don't allow us to detect whether we're getting the data from the host or containerized procfs - but that's fine for now, we'll revisit once we get to the cgroup-based metrics.